File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        seen={0:1}
        prefix,count=0,0
        remove=0
        for i in range(len(nums)):
            prefix+=nums[i]
            remove=prefix-k
            count+=seen.get(remove,0)

            seen[prefix]=seen.get(prefix,0)+1
        return count

        
                    

            
                
        

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>mpp;
        mpp[0]=1;
        int prefix_sum=0,count=0;
        for (int i=0;i<nums.size();i++){
            prefix_sum+=nums[i];
            int remove=prefix_sum - k;
            count+=mpp[remove];
            mpp[prefix_sum]+=1;
        }
        return count;

    }
};